### DISCUSSION TITLE
LED flicker and not working. Common ground problem?

### DISCUSSION THREAD

**Discussion by DariBer** [awawa-dev/HyperHDR#344] [Q&A] [answered]
I have installed HyperHDR on RPi4, I have also flashed HyperSerialESP8266, I created the first led config with the diod. 

I get some weird settings flicker/flashing LED, the effects are also not what I expected. 

I have one PS to my RPi and one PS to the LED. The ESP8266 is powered by the USB from the RPi4. Could it be a problem that I don’t have any common ground between the LED, ESP8266 and the RPi?

if that could be the problem, how can I solve the problem? Can I just put a cable from ground to all stuff, like take a cable from the ground from the LED PS, to the ES8266 and then to the RPi? Or will I get more problems?

**Comment by awawa-dev** [comment, marked as answer]
Post a video that shows the problem. Does the all the LEDs to flicker or it starts from specific LED in the strip? Most likely, the reason is the lack of common ground between the ESP8266 and the LED strip. One short cable is enough, you don't need second to include Rpi. Raspberry Pi doesn't drive your LED strip and beside Rpi & ESP already have common ground because are connected using USB.

**Reply by DariBer** [reply]
Yes, it was the common ground. Thanx for your help. 

Looks like LED configuration settings have changed since the guide was written at HyperHDR website. What would you recommend to set at "refresh time" and "Delay after connect". I have done some changes so not sure what's the default settings anymore.

**Reply by awawa-dev** [reply]
Set them both to zero.
